openghg/store/base/_base.py

```diff
@@ -176,16 +184,24 @@ def datasource_lookup(
         results = {}
         for key, _data in data.items():
             metadata = _data["metadata"]
-            required_metadata = {k.lower(): str(v).lower() for k, v in metadata.items() if k in required_keys}
+
+            def lower_if_string(val: Any) -> Any:
```

Comment:
Have a look at the [`to_lowercase` function](https://github.com/openghg/openghg/blob/devel/openghg/util/_strings.py#L71) within the openghg.util submodule (_strings.py file) to see if this covers this case.

If not, this functionality would be better off added there and imported to be used here.